packages/core-api/src/extensions/extensions.tsx

```diff
@@ -41,20 +41,28 @@ export function createRoutableExtension<
     component: {
       lazy: () =>
         component().then(InnerComponent => {
-          const RoutableExtensionWrapper = ((props: any) => {
+          const RoutableExtensionWrapper: any = (props: any) => {
             // Validate that the routing is wired up correctly in the App.tsx
             try {
               useRouteRef(mountPoint);
             } catch {
               throw new Error(
-                'Routable extension component was not discovered in the app element tree. ' +
+                `Routable extension component with mount point ${mountPoint} was not discovered in the app element tree. ` +
                   'Routable extension components may not be rendered by other components and must be ' +
                   'directly available as an element within the App provider component.',
               );
             }
             return <InnerComponent {...props} />;
-          }) as T;
-          return RoutableExtensionWrapper;
+          };
+
+          const componentName =
+            (InnerComponent as { displayName?: string }).displayName ||
+            InnerComponent.name ||
+            'LazyComponent';
+
+          RoutableExtensionWrapper.displayName = `RoutableExtension(${componentName})`;
```

Comment:
Add mountPoint to the display name too maybe?

Reply:
Hmm, bit verbose tbh. Hopefully this is good enough but we can see if it still causes trouble